From ec223e68a8340bc5c037236edf49c33e70ed1861 Mon Sep 17 00:00:00 2001
From: Yan Markman <ymarkman@marvell.com>
Date: Wed, 6 Feb 2019 18:35:54 +0200
Subject: [PATCH 0967/1051] net: mvpp2: bm init with virt-rls
 zero-initialization

After restart the BM_VIRT_RLS_REG register contains a value and
should be initialized by zero.

Change-Id: Ib1b203f086f751a5d7d71c163a00708261a76f86
Signed-off-by: Yan Markman <ymarkman@marvell.com>
Reviewed-on: https://sj1git1.cavium.com/3597
Tested-by: sa_ip-sw-jenkins
Reviewed-by: Stefan Chulski <Stefan.Chulski@cavium.com>
[Kevin: The original patch got from Marvell sdk10.0_19.06]
Signed-off-by: Kevin Hao <kexin.hao@windriver.com>
---
 drivers/net/ethernet/marvell/mvpp2/mvpp2_main.c | 6 +++++-
 1 file changed, 5 insertions(+), 1 deletion(-)

diff --git a/drivers/net/ethernet/marvell/mvpp2/mvpp2_main.c b/drivers/net/ethernet/marvell/mvpp2/mvpp2_main.c
index dbc7f2725e1a..1b8b870662da 100644
--- a/drivers/net/ethernet/marvell/mvpp2/mvpp2_main.c
+++ b/drivers/net/ethernet/marvell/mvpp2/mvpp2_main.c
@@ -464,9 +464,13 @@ static int mvpp2_bm_pool_destroy(struct platform_device *pdev,
 static int mvpp2_bm_pools_init(struct platform_device *pdev,
 			       struct mvpp2 *priv)
 {
-	int i, err, size;
+	int i, err, size, cpu;
 	struct mvpp2_bm_pool *bm_pool;
 
+	/* Initialize Virtual with 0x0 */
+	for_each_present_cpu(cpu)
+		mvpp2_thread_write(priv, cpu, MVPP2_BM_VIRT_RLS_REG, 0x0);
+
 	/* Create all pools with maximum size */
 	size = MVPP2_BM_POOL_SIZE_MAX;
 	for (i = 0; i < MVPP2_BM_POOLS_NUM; i++) {
-- 
2.17.1

